src/spectral/utils/spectral-validator.js

```diff
@@ -118,6 +115,14 @@ function setupStaticAssets(staticAssets, defaultSpectralRulesetURI) {
   });
 }
 
+function addDataToSpectralConfig(content) {
+  const config = yaml.safeLoad(content);
+  config['rules']['ibm-sdk-operations']['then']['functionOptions']['schema'] = {
+    $ref: path.join(__dirname, '../rulesets/schemas/x-sdk-operations.json')
+  };
+  return yaml.safeDump(config);
```

Comment:
Could you do this substitution without converting from/to yaml?  It seems like you could just regex replace `'../rulesets`` with `_dirname + '../rulesets'` and get the same result.  What's more, if we have any new instances where we have to reference resources in this directory, we won't need additional code here.

Reply:
That's a great idea. I will implement it that way.